src/app/users/create-user/create-user.component.ts

```diff
@@ -0,0 +1,121 @@
+/** Angular Imports */
+import { Component, OnInit } from '@angular/core';
+import { FormGroup, FormBuilder, Validators } from '@angular/forms';
+import { Router, ActivatedRoute } from '@angular/router';
+
+/** Custom Services */
+import { UsersService } from '../users.service';
+
+/** Custom Validators */
+import { passwordsMustMatchValidator, RepeatPasswordErrorStateMatcher } from './passwords-must-match.validator';
+
+/**
+ * Create user component.
+ */
+@Component({
+  selector: 'mifosx-create-user',
+  templateUrl: './create-user.component.html',
+  styleUrls: ['./create-user.component.scss']
+})
+export class CreateUserComponent implements OnInit {
+
+  /** User form. */
+  userForm: FormGroup;
+  /** Offices data. */
+  officesData: any;
+  /** Roles data. */
+  rolesData: any;
+  /** Staff data. */
+  staffData: any;
+  /** Repeat password matcher */
+  repeatPasswordMatcher = new RepeatPasswordErrorStateMatcher;
+
+  /**
+   * Retrieves the offices and roles data from `resolve`.
+   * @param {FormBuilder} formBuilder Form Builder.
+   * @param {UsersService} UsersService Users Service.
+   * @param {ActivatedRoute} route Activated Route.
+   * @param {Router} router Router for navigation.
+   */
+  constructor(private formBuilder: FormBuilder,
+              private usersService: UsersService,
+              private route: ActivatedRoute,
+              private router: Router) {
+    this.route.data.subscribe((data: {
+        offices: any,
+        roles: any
+      }) => {
+        this.officesData = data.offices;
+        this.rolesData = data.roles;
+      });
+  }
+
+  /**
+   * Creates and sets user form.
+   */
+  ngOnInit() {
+    this.createUserForm();
+    this.setUserForm();
+  }
+
+  /**
+   * Creates the user form.
+   */
+  createUserForm() {
+    this.userForm = this.formBuilder.group({
+      'username': ['', Validators.required],
+      'email': ['', [Validators.required, Validators.email]],
+      'firstname': ['', [Validators.required, Validators.pattern('(^[A-z]).*')]],
+      'lastname': ['', [Validators.required, Validators.pattern('(^[A-z]).*')]],
+      'sendPasswordToEmail': [true],
+      'passwordNeverExpires': [false],
+      'password': [''],
+      'repeatPassword': [''],
+      'officeId': ['', Validators.required],
+      'staffId': [''],
+      'roles': ['', Validators.required]
+    }, { validator: passwordsMustMatchValidator });
+  }
+
+  /**
+   * Sets the user form
+   */
+  setUserForm() {
```

Comment:
setConditionalControls()

Lets follow a similar approach to what is implemented here: https://github.com/openMF/web-app/blob/master/src/app/products/loan-products/loan-product-stepper/loan-product-terms-step/loan-product-terms-step.component.ts